File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/adapterlistener/CategoryListener.java
package cz.muni.fi.umimecesky.adapterlistener;

import android.util.Log;
import android.view.View;
import android.widget.CheckBox;

import cz.muni.fi.umimecesky.R;

class CategoryListener implements View.OnClickListener {

    private CategoryAdapter categoryAdapter;
    private CheckBox checkBox;

    CategoryListener(CategoryAdapter categoryAdapter, CheckBox checkBox) {
        this.categoryAdapter = categoryAdapter;
        this.checkBox = checkBox;
    }

    @Override
    public void onClick(View v) {
        int position;

        switch (v.getId()) {
            case R.id.categoryCheckLayout:
                CategoryAdapter.ViewHolder holder = ((CategoryAdapter.ViewHolder) v.getTag());
                position = (int) holder.checkBox.getTag();
                break;
            case R.id.checkBox:
                position = (int) v.getTag();
                break;
            default:
                throw new IllegalArgumentException("Wrong class: " + v.getClass().getName());
        }
        boolean changeChecked = !categoryAdapter.isChecked(position);
        categoryAdapter.setCategoryChecked(position, changeChecked);
        checkBox.setChecked(changeChecked);

        Log.v("checkbox" + position, String.valueOf(changeChecked));
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/adapterlistener/CategoryListener.kt
package cz.muni.fi.umimecesky.adapterlistener

import android.util.Log
import android.view.View
import android.widget.CheckBox

import cz.muni.fi.umimecesky.R

internal class CategoryListener(private val categoryAdapter: CategoryAdapter, private val checkBox: CheckBox) : View.OnClickListener {

    override fun onClick(v: View) {
        val position: Int

        when (v.id) {
            R.id.categoryCheckLayout -> {
                val holder = v.tag as CategoryAdapter.ViewHolder
                position = holder.checkBox.tag as Int
            }
            R.id.checkBox -> position = v.tag as Int
            else -> throw IllegalArgumentException("Wrong class: " + v.javaClass.name)
        }
        val changeChecked = !categoryAdapter.isChecked(position)
        categoryAdapter.setCategoryChecked(position, changeChecked)
        checkBox.isChecked = changeChecked

        Log.v("checkbox" + position, changeChecked.toString())
    }
}
